src/core/core.datasetController.js

```diff
@@ -287,17 +404,275 @@ helpers.extend(DatasetController.prototype, {
 		});
 	},
 
+	/**
+	 * @private
+	 */
+	_parse: function(start, count) {
+		var me = this;
+		var chart = me.chart;
+		var meta = me._cachedMeta;
+		var data = me._data;
+		var crossRef = chart._xref || (chart._xref = {});
+		var xScale = me._getIndexScale();
+		var yScale = me._getValueScale();
+		var xId = xScale.id;
+		var yId = yScale.id;
+		var xKey = getStackKey(xScale, yScale, meta);
+		var yKey = getStackKey(yScale, xScale, meta);
+		var stacks = xKey || yKey;
+		var i, ilen, parsed, stack, item, x, y;
+
+		if (helpers.isArray(data[start])) {
+			parsed = me._parseArrayData(meta, data, start, count);
+		} else if (helpers.isObject(data[start])) {
+			parsed = me._parseObjectData(meta, data, start, count);
+		} else {
+			parsed = me._parsePrimitiveData(meta, data, start, count);
+		}
+
+		function storeStack(stackKey, indexValue, scaleId, value) {
+			if (stackKey) {
+				stackKey += '.' + indexValue;
+				item._stackKeys[scaleId] = stackKey;
+				stack = crossRef[stackKey] || (crossRef[stackKey] = {});
+				stack[meta.index] = value;
+			}
+		}
+
+		for (i = 0, ilen = parsed.length; i < ilen; ++i) {
+			item = parsed[i];
+			meta.data[item._index]._parsed = item;
```

Comment:
I think if you make this line `meta.data[start + i]._parsed = item;` then you no longer need to set `_index` in the parsed data

Reply:
done